Criterion/Report.hs

```diff
@@ -103,6 +104,22 @@ report reports = do
     tpl <- loadTemplate [td,"."] template
     TL.writeFile name =<< formatReport reports tpl
 
+-- | Escape JSON string aimed to be embedded in an HTML <script> tag.
+-- Notably < and > are replaced with their unicode escape sequences such
+-- that closing the <script> tag from within the JSON data is disallowed.
+escapeJSON :: Char -> TL.Text
```

Comment:
Out of curiosity, where did you get the implementation for this function? I've seen various iterations of this function before, most notably in GHCJS ([here](https://github.com/ghcjs/ghcjs/blob/718b37fc7167269ebca633914c716c3dbe6d0faf/src/Compiler/JMacro/Base.hs#L887-L905)). The GHCJS version has quite a few differences from the code shown here, and I wonder if these are important.

Reply:
The GHCJS escaping seems to escape the JSON with respect to other JSON (or maybe JavaScript), but I'm trying to escape the JSON with respect to HTML here.

I started out with a normal XML escape using [`&<control>;` character references](https://html.spec.whatwg.org/multipage/syntax.html#character-references) from the provided mustache library, escaping the content in the attribute of the div tag originally used for embedding the JSON data. However, with this approach I was able to exit the quotes wrapping the JSON data and inject .

Then I tried to escape this attribute by writing a small escape sequence for the strings but given that both string and HTML escaping was necessary I went with the approach of just directly embedding the data in a <script> tag.

With this approach I once again tried with HTML escaping but this time those escape sequences were left intact after reading the data from the DOM. Thus I turned to escaping with [JSON unicode sequences](https://tools.ietf.org/html/rfc8259#section-7) instead. The most obvious character to escape is `<` to avoid parsing [end tags](https://html.spec.whatwg.org/multipage/syntax.html#end-tags) (more specifically `<script>`), however when looking into the problem I found that using `+` and [UTF-7 escape codes](https://en.wikipedia.org/wiki/UTF-7#Security) you can also trick some browsers to render a script tag. The unicode escape sequence approach is safe to use since after applying `JSON.parse` you get back exactly the code point the sequence resolves to which is the same character as we started with in Haskell. However, since HTML doesn't resolve these sequences and the `<script>` tag is listed as [raw text](https://html.spec.whatwg.org/multipage/syntax.html#raw-text-elements), it uses only uses ampersand for escaping there's no issue with the HTML parser treating the JSON as HTML data.

In conclusion we only _really_ need to escape `<`, `&`, and `+` inside a `<script>` tag for a standards compliant browser but given all inconsistencies in implementations I figured it might be wise to escape a bit more than strictly necessary. Things like null characters trip up a lot of implementations in weird places. Note that I didn't quote `"` since that would unnecessarily result in a lot of extra bytes in the output.

Reply:
I found a link to a Python library containing a helper function with the same goal: 

https://github.com/pallets/jinja/blob/9242a027c926a4636d33bac95b1f7654d8003ac9/src/jinja2/utils.py#L539

Reply:
This is what `Rails` does in their `json_escape` function:
https://github.com/rails/rails/blob/1165401ee962aef0aaf81080e3e7dcab522efa40/activesupport/lib/active_support/core_ext/string/output_safety.rb#L10
https://github.com/rails/rails/blob/1165401ee962aef0aaf81080e3e7dcab522efa40/activesupport/lib/active_support/core_ext/string/output_safety.rb#L113

Reply:
> The GHCJS escaping seems to escape the JSON with respect to other JSON (or maybe JavaScript), but I'm trying to escape the JSON with respect to HTML here.

Ah, that's an important distinction that I didn't realize.

> In conclusion we only _really_ need to escape `<`, `&`, and `+` inside a `<script>` tag for a standards compliant browser but given all inconsistencies in implementations I figured it might be wise to escape a bit more than strictly necessary. Things like null characters trip up a lot of implementations in weird places. Note that I didn't quote `"` since that would unnecessarily result in a lot of extra bytes in the output.

Gotcha. Since this part wasn't obvious to me until you pointed it out, could you include some version of this in the Haddocks/comments?